cli/pcluster/models/hit/hit_cluster_model.py

```diff
@@ -0,0 +1,124 @@
+    def test_configuration(self, pcluster_config):
+        """Try to launch the requested instances (in dry-run mode) to verify configuration parameters."""
+        cluster_section = pcluster_config.get_section("cluster")
+        vpc_section = pcluster_config.get_section("vpc")
+
+        if not cluster_section or cluster_section.get_param_value("scheduler") == "awsbatch" or not vpc_section:
+            return
+
+        master_instance_type = cluster_section.get_param_value("master_instance_type")
+
+        # Retrieve network parameters
+        compute_subnet = vpc_section.get_param_value("compute_subnet_id")
+        master_subnet = vpc_section.get_param_value("master_subnet_id")
+        vpc_security_group = vpc_section.get_param_value("vpc_security_group_id")
+        if not compute_subnet:
+            compute_subnet = master_subnet
+        security_groups_ids = []
+        if vpc_security_group:
+            security_groups_ids.append(vpc_security_group)
+
+        # Initialize CpuOptions
+        disable_hyperthreading = cluster_section.get_param_value("disable_hyperthreading")
+        master_instance_type_info = get_instance_type(master_instance_type)
+
+        # Set vcpus according to queue's disable_hyperthreading and instance features
+        vcpus_info = master_instance_type_info.get("VCpuInfo")
+        master_vcpus = vcpus_info.get("DefaultVCpus")
+
+        master_cpu_options = {"CoreCount": master_vcpus // 2, "ThreadsPerCore": 1} if disable_hyperthreading else {}
+
+        try:
+            latest_alinux_ami_id = self._get_latest_alinux_ami_id()
+
+            # Test Master Instance Configuration
+            self._ec2_run_instance(
+                pcluster_config,
+                InstanceType=master_instance_type,
+                MinCount=1,
+                MaxCount=1,
+                ImageId=latest_alinux_ami_id,
+                SubnetId=master_subnet,
+                SecurityGroupIds=security_groups_ids,
+                CpuOptions=master_cpu_options,
+                DryRun=True,
+            )
+
+            for _, queue_section in pcluster_config.get_sections("queue").items():
+                queue_placement_group = queue_section.get_param_value("placement_group")
+                queue_placement_group = (
+                    {"GroupName": queue_placement_group}
+                    if queue_placement_group not in [None, "NONE", "DYNAMIC"]
+                    else {}
+                )
+
+                compute_resource_settings = queue_section.get_param_value("compute_resource_settings")
+                for label in compute_resource_settings.split(","):
+                    compute_resource_section = pcluster_config.get_section("compute_resource", label)
+
+                    self.__test_compute_resource(
+                        pcluster_config,
+                        compute_resource_section,
+                        disable_hyperthreading=queue_section.get_param_value("disable_hyperthreading"),
```

Comment:
should we use `or disable_hyperthreading` already taken from the cluster or the queue?

Reply:
No. During this phase all the parameters in ``compute_resource`` must already have been processed and initialized by refresh operations.